### PR TITLE
Small stuff

### PR DESCRIPTION
**Pull request by tomleb**

### CODE REVIEW

**Comment by Cyberunner23** [comment]
The api url is fine actually, its the tests that broke when i forgot to update them along with the change in the url...

File: src/client.rs [hunk 1]
```diff
@@ -139,47 +140,22 @@ impl Client {
...
 
-        Ok(response)
+    pub fn post(&self, endpoint: String, header: Option<Headers>) -> Result<Response, error::Error> {
+        self.make_request(endpoint, header, Method::Post)
     }
 
-    //GET with a body
-    pub fn get_body(&mut self, endpoint: String, header: Option<Headers>, body: String) -> Result<Response, error::Error> {
 
+    fn make_request(&self, endpoint: String, header: Option<Headers>, method: Method) -> Result<Response, error::Error> {
```
Comment (Cyberunner23) [inline src/client.rs]: make_request_body has method at the beginning, can we keep it consistent?

File: src/client.rs [hunk 2]
```diff
@@ -108,11 +106,9 @@ impl Client {
 
     pub fn response_to_string(response: &mut Response) -> Result<String, error::Error> {
         let mut body_data = String::new();
-        match response.read_to_string(&mut body_data) {
-            Ok(_)    => (),
-            Err(err) => return Err(error::Error::STDIO(err))
-        }
-        return Ok(body_data);
+        response.read_to_string(&mut body_data)
+                .map(|_| body_data)
```
Comment (Cyberunner23) [inline src/client.rs]: Do we need map here?

Comment (tomleb) [inline src/client.rs]: Yeah we do, else we get a mismatched type error
    = note: expected type `std::result::Result<std::string::String, error::Error>`
    = note:    found type `std::result::Result<usize, error::Error>`

**Pull request merged**